### DISCUSSION TITLE
Adding a cover page to a manuscript

### DISCUSSION THREAD

**Discussion by awqk** [vkbo/novelWriter#1975] [Help] [answered]
My process is: generate standard markdown with the build tool, convert this to odt with pandoc using an odt reference file with a cover page and the desired styles. In the novelwriter text, only markdown formatting is available.

I have also considered generating odt with the build tool, using libreoffice to apply the styles of another odt file afterward. Preliminary tests look good, and this should be doable after some more fiddling around. This enables using all novelwriter formatting.

Both these options look a bit roundabout, though. I prefer some latex-like run-this-command-and-everything-will-be-done way to build the manuscript. So I tried to write the cover page in novelwriter. I'm running into trouble though, because I always disable "Preserve Hard Line Breaks" in the build options for the manuscript, whereas in the cover page (and only there) I would want to place line breaks exactly where I want them.

What are your thoughts about this?

**Comment by vkbo** [comment]
First of all, any particular reason why you turn off the line breaks?

Secondly, maybe a shortcode for absolutely-always-put-a-line-break-here would be useful for such cases. Maybe simply `[br]`.

**Reply by awqk** [reply]
> First of all, any particular reason why you turn off the line breaks?

I was anticipating this question:
- it's the markdown way
- it sometimes simplifies editing if sentences start on a new line, or if a long paragraph is broken down into several lines
- better diff output when reviewing changes.

> Secondly, maybe a shortcode for absolutely-always-put-a-line-break-here would be useful for such cases. Maybe simply `[br]`.

Do we have a feature request here?
RFC 7764 lists a few markdown standards. How about using pandoc's `escaped_line_breaks` extension, ending a line with `\`?
In any case it's probably not as simple as it looks, it still has to work with other format modifiers, like `>>` `<<` etc.

**Reply by vkbo** [reply]
novelWriter used to support Markdown standard double space line ending, but it was really hard to see, so it was removed at some point. I prefer not to add anything new non-standard formatting that isn't on shortcode format. I don't want it to become a chaos of styles.

As for paragraph alignment, that already deals with line breaks in the regular case. You have to apply them to the first line in a multi-line paragraph. You can't have multiple different alignments within a single paragraph. Not even Libre Office allows that.

**Reply by awqk** [reply]
> novelWriter used to support Markdown standard double space line ending, but it was really hard to see, so it was removed at some point. I prefer not to add anything new non-standard formatting that isn't on shortcode format. I don't want it to become a chaos of styles.

I agree about the double blanks, it has always been a silly idea. Shortcode or backslash, I think it will be useful.

> As for paragraph alignment, that already deals with line breaks in the regular case. You have to apply them to the first line in a multi-line paragraph. You can't have multiple different alignments within a single paragraph. Not even Libre Office allows that.

I know, I was referring to the implementation side of things. `<<` must be recognized even when it's not at the end of a line (because a mandatory line break might follow). I am also looking at the user guide section 9.10.x, and see possible weirdness.

Now that you mention it, you can have multiple alignments in a libreoffice paragraph, using left/right/centered tabs.

**Reply by vkbo** [reply]
> I know, I was referring to the implementation side of things. `<<` must be recognized even when it's not at the end of a line (because a mandatory line break might follow). I am also looking at the user guide section 9.10.x, and see possible weirdness.

No, that is not necessary as the alignment still applies to the whole paragraph. A hard line break does not create a new paragraph, as is the case in Libre Office.

The parser does consider the text line by line, so I made a choice to only keep the alignment information for the first line in a paragraph. That's just a parsing choice, and the easiest one.

**Comment by peter88213** [comment, marked as answer]
> I have also considered generating odt with the build tool, using libreoffice to apply the styles of another odt file afterward. Preliminary tests look good, and this should be doable after some more fiddling around. This enables using all novelwriter formatting.
>
> Both these options look a bit roundabout, though. I prefer some latex-like run-this-command-and-everything-will-be-done way to build the manuscript. 

Have you already considered creating a master document for LibreOffice (odm) that includes the frontmatter, backmatter and the document exported by novelWriter?

**Reply by awqk** [reply]
Actually, I wasn't aware this LibreOffice feature exists, but it sounds like the obvious solution.

In the meantime, I have spent (wasted) some more time on loading styles. You can get the cover page inserted if the cover page is embedded in the first page header of the template file.

It's probably no surprise to serious LibreOffice users that loading styles from another odt file also overrides just about every novelwriter build setting, including first line indent, page headers or footers, paragraph spacing etc. You can use this to change build settings not implemented in novelwriter. Create a template once (for example by changing the novelwriter build of your manuscript) and load the styles from this template after every build.

**Reply by peter88213** [reply]
> In the meantime, I have spent (wasted) some more time on loading styles. You can get the cover page inserted if the cover page is embedded in the first page header of the template file.

Actually, I think experimenting with templates and styles is useful in every respect. It's the right way to get the most out of LibreOffice Writer, and experience never hurts. On the other hand, as the name suggests, templates should of course be universally applicable and reusable. 
It is interesting to note that the styles used by a master document automatically apply to all included documents, even if these are based on other templates as individual files. 

For me, it is essential to avoid all "hard" formatting and to design my document exclusively with paragraph and character styles. You can also achieve this afterwards with macros. I have created such a macro as an extension for the conversion between "italic/bold" and "emphasized/strongly emphasized": https://peter88213.github.io/emph/

If you want to keep experimenting with templates, another extension that I use for my everyday work might come in handy: https://peter88213.github.io/StyleSwitcher/

The possibilities offered by LibreOffice's template documents can be seen in the 5x8-inch book template that I created a few years ago in a different context. The workflow described there is probably irrelevant for novelWriter users, but the template itself and the font might be worth a look: https://github.com/peter88213/yw2html/tree/main/add-on

One last note on the book cover: My preferred book-on-demand service provider requires the book cover and the book block to be separate PDF files. I therefore design the book cover according to the format specifications using scribus and Photoshop.

**Comment by vkbo** [comment]
Just a heads up that PR #2063 adds a new shortcode that forces the line break regardless of the preserve line breaks flag in the Manuscript Tool. You can add `[br]` to the text (available from the insert menu also).

It will consider the `[br]` tag and the first following line break, as a forced break. The latter means you can add `[br]\n`, and both have it visually correct in the editor, and in the output.

I've also changed the rules for how the indentation and alignment tags are processed. They can be added on any line of the text, but if they follow a `[br]` tag they will be ignored. That's a side-effect of the parsing. However, this will work:

```
I am the very model of a modern Major-General[br]
I've information vegetable, animal, and mineral[br]
I know the kings of England, and I quote the fights historical[br]
From Marathon to Waterloo, in order categorical <<
```

This text will always be broken into lines and left-aligned regardless of line break settings and default alignment and first line indentation.

It should also be suitable for adding author information to a cover page. In 2.6 it will also be possible to turn off the automatic page break before the `#!` title, so you can put contact details above it:

```
Lisa Simpson[br]
742 Evergreen Terrace[br]
12345 Springfield <<

[vspace:5]

#! My Novel

>> **By Lisa Simpson** <<
```

I believe this will solve  most use cases.
